src/games/supported/DoubleDunk.cpp

```diff
@@ -116,3 +118,49 @@ ActionVect DoubleDunkSettings::getStartingActions() {
     startingActions.push_back(PLAYER_A_UPFIRE);
     return startingActions;
 }
+
+// returns a list of mode that the game can be played in
+ModeVect DoubleDunkSettings::getAvailableModes(){
+    // this game has a menu that allows to define various yes/no options
+    // setting these options define in a way a different mode
+    // there are 4 relevant options, which makes 2^4=16 available modes
+    ModeVect modes(16);
+    for(unsigned i = 0; i < 16; i++){
+        modes[i] = i;
+    }
+    return modes;
+}
+
+// set the mode of the game
+// the given mode must be one returned by the previous function
+void DoubleDunkSettings::setMode(game_mode_t m, System &system, StellaEnvironment& environment){
+    if(m < 16){ /*m >= 0 is implicit, since m is an unsigned int*/
+        m_mode = m;
+        //push the select button to open the menu
+        environment.pressSelect(1);
+        //discard the first two entries (irrelevant)
+        environment.act(PLAYER_A_DOWN, PLAYER_B_NOOP);
+        environment.act(PLAYER_A_NOOP, PLAYER_B_NOOP);
+        environment.act(PLAYER_A_DOWN, PLAYER_B_NOOP);
```

Comment:
Why not also make the game length an option? & the player colours? (yes, that's a lot of modes... but lots of opportunities for transfer?)